applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2137,9 +2137,9 @@ public function save($formPostValues, $settings = false) {
                     // Updating
                     $stored = $this->getID($discussionID, DATASET_TYPE_OBJECT);
 
-                    // Block Format change if we're forcing the formatter.
-                    if (c('Garden.ForceInputFormatter')) {
-                        unset($fields['Format']);
+                    // Make sure that the discussion get formatted in the method defined by Garden.
+                    if (!val('Format', $fields) || c('Garden.ForceInputFormatter')) {
```

Comment:
`val()` function is deprecated we use coalescing  operator instead
```suggestion
                    if (!($fields['Format'] ?? false) || c('Garden.ForceInputFormatter')) {
```